lib/_http_client.js

```diff
@@ -373,11 +373,6 @@ function socketCloseListener() {
   const req = socket._httpMessage;
   debug('HTTP socket close');
 
-  // Pull through final chunk, if anything is buffered.
-  // the ondata function will handle it properly, and this
-  // is a no-op if no final chunk remains.
-  socket.read();
```

Comment:
readable is false here, hence `read` is not "safe" to call. The test for the bug this is supposed to fix passes without this (very old bug fix).